Controller/HealthCheckController.php

```diff
@@ -11,17 +12,19 @@
 
 class HealthCheckController
 {
+    protected $container;
     protected $runner;
     protected $pathHelper;
     protected $template;
 
     /**
-     * @param Runner     $runner
-     * @param PathHelper $pathHelper
+     * @param ContainerInterface $container
+     * @param PathHelper         $pathHelper
+     * @param                    $template
      */
-    public function __construct(Runner $runner, PathHelper $pathHelper, $template)
+    public function __construct(ContainerInterface $container, PathHelper $pathHelper, $template)
     {
-        $this->runner = $runner;
+        $this->container = $container;
```

Comment:
instead of injecting the container it might be nicer to create a ``manager`` service that in turn gets the container injected and that contains the logic for fetching runners from the container .. this way the logic is more encapsulated from the controller.